src/Microsoft.Health.SqlServer/Features/Health/SqlServerHealthCheck.cs

```diff
@@ -42,7 +45,9 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
         _logger.LogInformation($"Starting {nameof(SqlServerHealthCheck)}.");
 
         CustomerKeyHealth cmkStatus = await _customerKeyHealthCache.GetAsync(cancellationToken).ConfigureAwait(false);
-        if (!cmkStatus.IsHealthy)
+
+        if (!cmkStatus.IsHealthy &&
```

Comment:
Something still smells funny about this. Essentially every health check that relies on CMK makes the same check with some modifications. I would say either:
1. We use some sort of abstract class for these health checks that check CMK to perform the preconditions
2. Manage the relationships between health checks outside of their implementations ala some sort of decorator pattern

Reply:
I agree it was a little weird - refactored it a bit to make "dependent" classes. `KeyAccessDependent` and `KeyAccessAndDataStoreStateDependent`, which define which reasons that they care about. The health checks can inherit this (and possibly other things in the future? could help with exception handling) and use it to decide whether or not to continue after getting the CMK health.
LMK what you think